<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>485</width>
    <height>410</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QComboBox" name="BaudBox">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>36</y>
      <width>62</width>
      <height>20</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>9600</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>19200</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>38400</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>57600</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>115200</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>1500000</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="PortBox">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>10</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="BitBox">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>62</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>8</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="StopBox">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>114</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>1</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="labelSerial">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>36</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>串口：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="ParityBox">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>88</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>0</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="labelParityBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>88</y>
      <width>48</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>校验位：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OpenSerialButton">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>150</y>
      <width>61</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>打开串口</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelStopBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>114</y>
      <width>48</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>停止位：</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelBaudBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>36</y>
      <width>48</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>波特率：</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelBitBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>62</y>
      <width>48</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>数据位：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="SearchSerialButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>61</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>查找串口</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="InputLine1">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>110</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="labelInput1">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>110</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>输入值：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="OutputLine1">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>150</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="labelOutput1">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>150</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>输出值：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="InputLine2">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>290</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="labelOutput2">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>330</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>输出值：</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelInput2">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>290</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>输入值：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="OutputLine2">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>330</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="labelSelectAlgo">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>210</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>算法选择：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="AlgoSelectBox">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>210</y>
      <width>62</width>
      <height>20</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>DES</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>3DES</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>SM2</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="labelSelectMode">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>250</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>模式选择：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="ModeSelectBox">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>250</y>
      <width>62</width>
      <height>20</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>ECB</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>CBC</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="SwitchSafeModButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>190</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>打开加密模块</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cerSaveButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>30</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>保存证书</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cerReadButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>70</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>读取证书</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cerDeleteButton">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>30</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>删除证书</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cerDelAllButton">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>70</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>删除全部证书</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>10</y>
      <width>311</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>证书相关模块</string>
    </property>
    <widget class="QPushButton" name="SM2SignButton">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>20</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>SM2签名</string>
     </property>
    </widget>
    <widget class="QPushButton" name="SM2CheckButton">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>60</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>SM2验签</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>190</y>
      <width>311</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>数据加解密模块</string>
    </property>
    <widget class="QPushButton" name="encryptionButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>数据加密</string>
     </property>
    </widget>
    <widget class="QPushButton" name="decryptionButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>数据解密</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelKeyNum">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>20</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>密钥索引：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="LineKeyNum">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>60</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="KeyDownloadButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>220</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>公钥下载</string>
    </property>
   </widget>
   <widget class="QPushButton" name="CADownloadButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>310</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>CA证书下载</string>
    </property>
   </widget>
   <widget class="QPushButton" name="GenelKeyButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>340</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>公私钥对生成</string>
    </property>
   </widget>
   <widget class="QPushButton" name="KeyDownloadButton_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>250</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>私钥下载</string>
    </property>
   </widget>
   <widget class="QPushButton" name="KekDownloadButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>280</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>私钥下载</string>
    </property>
   </widget>
   <zorder>groupBox_2</zorder>
   <zorder>groupBox</zorder>
   <zorder>BaudBox</zorder>
   <zorder>PortBox</zorder>
   <zorder>BitBox</zorder>
   <zorder>StopBox</zorder>
   <zorder>labelSerial</zorder>
   <zorder>ParityBox</zorder>
   <zorder>labelParityBox</zorder>
   <zorder>OpenSerialButton</zorder>
   <zorder>labelStopBox</zorder>
   <zorder>labelBaudBox</zorder>
   <zorder>labelBitBox</zorder>
   <zorder>SearchSerialButton</zorder>
   <zorder>InputLine1</zorder>
   <zorder>labelInput1</zorder>
   <zorder>OutputLine1</zorder>
   <zorder>labelOutput1</zorder>
   <zorder>InputLine2</zorder>
   <zorder>labelOutput2</zorder>
   <zorder>labelInput2</zorder>
   <zorder>OutputLine2</zorder>
   <zorder>labelSelectAlgo</zorder>
   <zorder>AlgoSelectBox</zorder>
   <zorder>labelSelectMode</zorder>
   <zorder>ModeSelectBox</zorder>
   <zorder>SwitchSafeModButton</zorder>
   <zorder>cerSaveButton</zorder>
   <zorder>cerReadButton</zorder>
   <zorder>cerDeleteButton</zorder>
   <zorder>cerDelAllButton</zorder>
   <zorder>KeyDownloadButton</zorder>
   <zorder>CADownloadButton</zorder>
   <zorder>GenelKeyButton</zorder>
   <zorder>KeyDownloadButton_2</zorder>
   <zorder>KekDownloadButton</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>485</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
